src/tasks/tasks.service.ts

```diff
@@ -1,26 +1,132 @@
-import { Injectable } from '@nestjs/common';
+import { HttpStatus, Injectable } from '@nestjs/common';
 import { CreateTaskDto } from './dto/create-task.dto';
 import { UpdateTaskDto } from './dto/update-task.dto';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Response } from 'express';
+import { paginate } from 'nestjs-typeorm-paginate';
+import { Task } from 'src/tasks/entities/task.entity';
+import { PaginationDto } from 'src/common/dto/pagination.dto';
+import { BaseResponse } from 'src/utils/base-response.utils';
+import { I18nService } from 'nestjs-i18n';
+import { DEFAULT_PAGE, DEFAULT_PER_PAGE } from 'src/constant/app-constant';
+import handleDatabaseOperation from 'src/utils/handle-database-async';
+import { User } from 'src/auth/entities/user.entity';
+import { findTaskById, TaskQueryType } from 'src/utils/task.utils';
+import { findUserById, UserQueryType } from 'src/utils/user.utils';
 
 @Injectable()
 export class TasksService {
-  create(createTaskDto: CreateTaskDto) {
-    return 'This action adds a new task';
+  constructor(
+    @InjectRepository(Task)
+    private readonly taskRepository: Repository<Task>,
+
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+
+    private readonly i18n: I18nService,
+  ) {}
+
+  async create(lang: string, res: Response, createTaskDto: CreateTaskDto) {
+    const userId = createTaskDto.userId;
+    const user = (await findUserById(
+      userId,
+      UserQueryType.BASIC,
+      res,
+      await this.i18n.translate('user.error.notFound', { lang }),
+    )) as User;
```

Comment:
```suggestion
    ));
```
Bỏ as User này, tương tự những chỗ khác